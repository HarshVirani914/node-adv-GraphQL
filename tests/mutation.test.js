const gql = require("graphql-tag");
const createTestServer = require("./helper");

const CREATE_POST = gql`
  mutation {
    createPost(input: { message: "hello" }) {
      message
    }
  }
`;

const UPDATE_SETTINGS = gql`
  mutation {
    updateSettings(input: { theme: DARK }) {
      theme
    }
  }
`;

describe("mutations", () => {
  test("createPost", async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          createOne: jest.fn(() => ({
            message: "hello",
          })),
        },
      },
    });

    const res = await mutate({ mutation: CREATE_POST });
    expect(res).toMatchSnapshot();
  });

  test("updateSettings", async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Settings: {
          updateOne: jest.fn(() => ({
            theme: "DARK",
          })),
        },
      },
    });

    const res = await mutate({ mutation: UPDATE_SETTINGS });
    expect(res).toMatchSnapshot();
  });
});
