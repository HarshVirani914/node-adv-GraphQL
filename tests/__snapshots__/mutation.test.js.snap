// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations createPost 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "message": "hello",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutations updateSettings 1`] = `
Object {
  "data": Object {
    "updateSettings": Object {
      "theme": "DARK",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
